alerts/cluster-monitoring-operator/PrometheusOperatorRejectedResources.md

```diff
@@ -0,0 +1,77 @@
+# PrometheusOperatorRejectedResources
+
+## Meaning
+
+Custom Resources managed by Prometheus Operator were rejected and not propagated
+to operated Prometheus and Alertmanager.
+The alert `PrometheusOperatorRejectedResources` is triggered by misconfiguration
+of these custom resources:
+- Alertmanager
+- PodMonitor
+- Probe
+- PrometheusRule
+- ScrapeConfig
+- ServiceMonitor
+
+## Impact
+
+The custom resource triggering the alert will be ignored by Prometheus Operator,
+and by consequence will not be integrated into the configuration of the Prometheus
+and the Alertmanager managed by Prometheus Operator.
+
+## Diagnosis
+
+Find out the type of custom resource and its namespace in the `description` field
+of the alert `PrometheusOperatorRejectedResources`.
+
+By CLI
+```bash
+### Gets alertmanager URL
+$ ALERT_MANAGER=$(oc get route alertmanager-main -n openshift-monitoring -o jsonpath='{@.spec.host}')
+
+### Gets fired alerts
+$ curl -k -H "Authorization: Bearer $(oc create token prometheus-k8s -n openshift-monitoring)"  https://$ALERT_MANAGER/api/v1/alerts
+```
+
+In Openshift web console:
+Browse to the tab Observe -> Alerting, then search the alert `PrometheusOperatorRejectedResources`
+in firing alerts list.
+![console alert list](PrometheusOperatorRejectedResources-alert-list.png)
+
+Click on the alert to see its details and scroll down to see its `Description` field.
+The namespace should be displayed as a link, followed by the resource type.
+
+![console alert detail](PrometheusOperatorRejectedResources-alert-detail.png)
+
+Then check recently modified custom resource of that type in that namespace.
+
+For more details of the error, check logs of Prometheus Operator pod in
+the namespace `openshift-user-workload-monitoring`.
+```bash
+# Get the name of the Prometheus Operator pod
+POD=$(oc get pod -n openshift-user-workload-monitoring | grep -Eo 'prometheus-operator-[0-9a-z]*-[0-9a-z]*')
+
+# Check its logs
+oc logs $POD -n openshift-user-workload-monitoring
+```
+
+Here is a typical error message about rejected custom resource.
+```log
+level=warn ts=2023-07-03T20:37:20.740723141Z caller=operator.go:1917 component=prometheusoperator msg="skipping servicemonitor" error="it accesses file system via bearer token file which Prometheus specification prohibits" servicemonitor=quarkus-demo/otel-collector namespace=openshift-user-workload-monitoring prometheus=user-workload
+```
+
+## Mitigation
+
+Fix the misconfigured custom resource and reapply it to the cluster.
+
+The custom resource itself should already conform CRD schema, elsewise the API server
+will reject it before the Prometheus Operator does.
```

Comment:
I suggest we tell users not to bother with the schema, the problem comes from value, because user may know that schema is validated before writing the resource to database.